video/out/wayland_common.c

```diff
@@ -1530,6 +1533,100 @@ static const struct wp_image_description_v1_listener image_description_listener
     image_description_failed,
     image_description_ready,
 };
+
+static void info_done(void *data, struct wp_image_description_info_v1 *image_description_info)
+{
+    wp_image_description_info_v1_destroy(image_description_info);
+}
+
+static void info_icc_file(void *data, struct wp_image_description_info_v1 *image_description_info,
+                          int32_t icc, uint32_t icc_size)
+{
+    struct vo_wayland_state *wl = data;
+    if (wl->icc_size) {
+        munmap(wl->icc_file, wl->icc_size);
+        wl->icc_file = NULL;
+        wl->icc_size = 0;
+    }
+    void *icc_file = mmap(NULL, icc_size, PROT_READ, MAP_PRIVATE, icc, 0);
```

Comment:
Are we sure it is safe to keep this file mmaped? It the underlying resource disappear in anyway we will crash with buserror on next use.

Reply:
It should be, if the underlying resource disappears we should receive the event again https://wayland.app/protocols/color-management-v1#wp_image_description_info_v1:event:icc_file

Reply:
Although I guess I should close the fd.